sdk/src/gateway/client.ts

```diff
@@ -197,13 +198,41 @@ export class GatewayApiClient {
                 data,
             };
         } else if (params.toChain.toString().toLowerCase() === 'bitcoin') {
+            const chain = getChainConfig(params.fromChain);
+            const publicClient = viemClient(chain);
             const data = await this.getOfframpQuote(params);
+
+            const [offrampOrder, offrampRegistryAddress, feeValues, gasPrice] = await Promise.all([
+                this.createOfframpOrder(data, params),
+                this.fetchOfframpRegistryAddress(),
+                publicClient.estimateFeesPerGas(),
+                publicClient.getGasPrice(),
+            ]);
+
+            const fee = feeValues.maxFeePerGas ?? gasPrice;
+
+            const createOrderGasEstimate = await publicClient.estimateContractGas({
+                address: offrampRegistryAddress,
+                abi: offrampOrder.offrampABI,
+                functionName: offrampOrder.offrampFunctionName,
+                args: offrampOrder.offrampArgs,
+                account: params.fromUserAddress as Address,
+            });
```

Comment:
Qs: say `fromUserAddress` doesn't have enough allowance, the `estimateContractGas` will fail right? 
We should do `stateOverride`